scanner/src/main/kotlin/ScannerWrapperFactory.kt

```diff
@@ -22,16 +22,26 @@ package org.ossreviewtoolkit.scanner
 import java.util.ServiceLoader
 
 import org.ossreviewtoolkit.utils.common.Options
-import org.ossreviewtoolkit.utils.common.Plugin
+import org.ossreviewtoolkit.utils.common.TypedConfigurablePluginFactory
 
 /**
  * A common abstract class for use with [ServiceLoader] that all [ScannerWrapperFactory] classes need to implement.
  */
-abstract class ScannerWrapperFactory<out T : ScannerWrapper>(override val type: String) : Plugin {
+abstract class ScannerWrapperFactory<CONFIG>(override val type: String) :
+    TypedConfigurablePluginFactory<CONFIG, ScannerWrapper> {
+    override fun create(options: Options, secrets: Options): ScannerWrapper {
+        val (matcherConfig, filteredOptions) = ScannerMatcherConfig.create(options)
+        return create(parseConfig(filteredOptions, secrets), matcherConfig)
+    }
+
+    final override fun create(config: CONFIG): ScannerWrapper {
+        throw UnsupportedOperationException("Use create(CONFIG, ScannerMatcherConfig) instead.")
```

Comment:
Are these backticks now instead of single quotes? If I'm not mistaken so far we never use backticks in such output, so I'd prefer to stick to single quotes.

Reply:
Right, I used backticks out of habit, switched to single quotes now.